src/ctelnet.cpp

```diff
@@ -1908,3 +1910,34 @@ void cTelnet::raiseProtocolEvent(const QString& name, const QString& protocol)
     event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
     mpHost->raiseEvent(event);
 }
+
+// credit: https://github.com/qflow/websockets
+void cTelnet::setKeepAlive(int socketHandle)
+{
+#ifdef Q_OS_WIN
+    struct tcp_keepalive
+    {
+        u_long onoff;
+        u_long keepalivetime;
+        u_long keepaliveinterval;
+    } alive;
+    alive.onoff = TRUE;
+    alive.keepalivetime = 2 * 60 * 1000;
+    alive.keepaliveinterval = 3000;
+    DWORD dwBytesRet = 0;
+    int res = WSAIoctl(socketHandle, SIO_KEEPALIVE_VALS, &alive, sizeof(alive), NULL, 0, &dwBytesRet, NULL, NULL);
+#endif
+#ifdef Q_OS_LINUX
```

Comment:
What is wrong with a fine `#elif defined(Q_OS_LINUX)` here {and perhaps used with an opening `#if defined (Q_OS_WIN32)`} at the start - it makes it clearer that you are expressing a *choose one block of code for your OS* selection here.

BTW since we are not targeting  the Windows Runtime (Windows Stores applications on Windows 8, Windows RT or Windows Phone 8) we should be using `Q_OS_WIN32` and **not** `Q_OS_WIN`.

Finally - what about `Q_OS_MACOS` and (my personal choice of primary OS) `Q_OS_FREEBSD` - there is nothing here for them!

Reply:
Fixed, and I don't know about FreeBSD networking code - by not officially supported I don't worry myself about it. Let me know what it should be for that!